### PR TITLE
Adds tests corresponding to issue 704. [DO NOT MERGE]

### PR DESCRIPTION
**Pull request by lemire**
This adds a few tests corresponding to the uint64_t, int64_t boundary.

### CODE REVIEW

**Comment by lemire** [comment]
This is just me being paranoid.

**Comment by jkeiser** [comment]
@lemire a lot of the edge tests are already addressed here: https://github.com/simdjson/simdjson/blob/c54c6f4b41d7b344376d59d98068687c8efe460d/tests/basictests.cpp#L1643-L1650

It definitely checks `is<uint64_t>() == false`; however, it does not check `get<uint64_t>()` throws an error ...

**Comment by lemire** [comment]
> It definitely checks is<uint64_t>() == false; however, it does not check get<uint64_t>() throws an error ... 

I do not mind dropping the current PR on the floor, but I do think we should check that exceptions are thrown.

**Comment by jkeiser** [comment]
> I do not mind dropping the current PR on the floor, but I do think we should check that exceptions are thrown.

Agreed. It just probably belongs in the type tests, I think--it's weird to have them split across methods. I'll take a look at it.

File: tests/basictests.cpp [hunk 1]
```diff
@@ -697,6 +702,62 @@ namespace dom_api_tests {
...
+    if(err == SUCCESS) {
+      cerr << "get<int64_t>() should fail" << endl;
+      return false;
+    }
+    auto [v2,er2] = obj["t"].get<uint64_t>();
+    if(er2 != SUCCESS) {
+      cerr << "get<uint64_t>() should work" << endl;
+      return false;
+    }
+    if(v2 != UINT64_C(9223372036854775808)) {
```
Comment (jkeiser) [inline tests/basictests.cpp:754]: I'm surprised this compiles given it doesn't fit into a ULL!

Comment (lemire) [reply tests/basictests.cpp:754]: 9223372036854775808 < (1<<64) is true.

Comment (jkeiser) [reply tests/basictests.cpp:754]: Never mind, it's a UL.

**Comment by lemire** [comment]
I have changed the title of this PR to reflect the fact that we do not want to merge it. But I would prefer to keep it around until we resolve the issue where we do not test that the value extraction functions throw.

**Comment by jkeiser** [comment]
I think #715 handles the intent of this.

**Comment by lemire** [comment]
Closing.

**Pull request closed without merging**